src/store.c

```diff
@@ -79,31 +85,59 @@ struct rill_store
 // -----------------------------------------------------------------------------
 
 static struct encoder store_encoder(
-        struct rill_store *store, struct indexer *indexer)
+        struct rill_store *store,
+        struct indexer *indexer,
+        struct vals* vals,
+        uint64_t offset)
 {
     return make_encoder(
-            store->vma + store->head->data_off,
-            store->vma + store->vma_len,
-            store->vals, indexer);
-}
-
-static struct decoder store_decoder(struct rill_store *store)
-{
-    return make_decoder(
-            store->vma + store->head->data_off,
+            store->vma + offset,
             store->vma + store->vma_len,
-            store->vals, store->index);
+            vals,
+            indexer);
 }
 
 static struct decoder store_decoder_at(
-        struct rill_store *store, size_t key_idx, uint64_t off)
+        struct rill_store *store,
+        size_t key_idx,
+        uint64_t curr_off,
+        enum rill_col column)
 {
+    assert(column == rill_col_a || column == rill_col_b);
```

Comment:
In internal function I wouldn't bother with asserting sanity on enums. Might be worth on the public interface but internally this should basically never be an issue that comes up.

Reply:
This is fair. I'll change this soon, thanks!